arelle/PluginManager.py

```diff
@@ -143,25 +143,37 @@ def close():  # close all loaded methods
 
 '''
 
+def logPluginTrace(message: str, error: bool=False) -> None:
+    """
+    If PLUGIN_TRACE_FILE is configured, logs `message` to it.
+    Otherwise, only prints to `stderr` if `error` flag is `True`
+    """
+    if PLUGIN_TRACE_FILE:
+        with open(PLUGIN_TRACE_FILE, "at", encoding='utf-8') as fh:
+            fh.write(message + '\n')
```

Comment:
suggest either breaking into 2 lines or using a formatted string to avoid an exception if message is `None`

Reply:
Also might be helpful to encode some general diagnostic info like time and thread ect. Suggest making a logger that either writes to a file or to std out instead of explicitly writing to a file. Maybe:

```python
import logging
import sys
...
logger = setup_logging(logging.StreamHandler(sys.stdout) if not PLUGIN_TRACE_FILE else logging.FileHandler(PLUGIN_TRACE_FILE))
...

def setup_logging(handler, formatter=logging.Formatter('%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')):
    handler.setLevel(PLUGIN_TRACE_LEVEL)
    handler.setFormatter(formatter)
    logger = logging.getLogger(f'{__name__})
    logger.addHandler(handler)
    return logger
...
def logPluginTrace(message: str, level) -> None:
    logger.log(message, level)
```